test/socket.cpp

```diff
@@ -291,6 +433,75 @@ BOOST_AUTO_TEST_CASE(accept_receive_send___connect_send_receive)
     ios.run();
 }
 
+BOOST_AUTO_TEST_CASE(accept___close)
```

Comment:
I don't have problem ditching the triple underscore for something else. But I don't think `then` is suitable in this case. If you have a look at the other tests, `then` would be more suited where single underscore is used. I'm using triple underscore to distinguish between "what different sides of the connection do" (if it makes sense).